### PR TITLE
[XABT] Break BuildApk into individual tasks for each content type.

### PR DESCRIPTION
**Pull request by jpobst**
Refactor `BuildApk` into individual tasks for each content type.  This promotes easier understanding of which pieces are dependent on each other and on which MSBuild properties.

At this point, several of these tasks (like `CollectAssemblyFilesForArchive`) are still unconditionally creating and modifying files on every run. A future PR will separate these creations/modifications into separate _targets_ that can be run incrementally.

Theoretically, this PR is likely a slight performance regression due to having more tasks and a little bit of duplicated logic, however this should be more than offset by the ability to run tasks incrementally in the future. (The performance difference is small enough that it isn't really measurable because it is within the variability of our build time.)

| _BuildApkEmbed/_BuildApkFastDev target | `main` | This PR |
|--------|--------|--------|
| Debug (FastDev) | 2.674s | 2.655s |
| Debug (EmbedAssembliesIntoApk = true) | 64.384s | 64.357s |
| Release | 6.130s | 6.094s |

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+		if (!ELFHelper.IsEmptyAOTLibrary (Log, item.filePath)) {
+			files.Add (item);
+		} else {
+			Log.LogDebugMessage ($"{item.filePath} is an empty (no executable code) AOT assembly, not including it in the archive");
+		}
+	}
+
+	// This method is used only for internal warnings which will never be shown to the end user, therefore there's
+	// no need to use coded warnings.
+	void LogSanitizerWarning (string message)
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs:296]: This is probably the original code, but what is a "internal warning"? This shows the warning to the user, how is it internal?

File: src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+using ArchiveFileList = System.Collections.Generic.List<(string filePath, string archivePath)>;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks;
+
+public class CollectNativeFilesForArchive : AndroidTask
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs:17]: The other ones had a summary, might be worth putting a quick one here, too.

**Review by jonathanpeppers — APPROVED** [review]
Overall looks good. 👍 

I don't think we are going to notice the overhead of adding more MSBuild tasks that split up the work. And it seems nice to split up all this code!

It's probably in the order of microseconds/nanoseconds difference.

File: src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs (hunk 1 above)
Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs:296]: Seems like this should just use `LogDebugMessage()`?

Comment (jpobst) [reply src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs]: It is original code, however it looks like it is only used for "checked" builds which requires the internal property `$(_AndroidCheckedBuild)` to be set.  I updated the comment on it to mention this.

File: src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs (hunk 2 above)
Comment (jpobst) [reply src/Xamarin.Android.Build.Tasks/Tasks/CollectNativeFilesForArchive.cs:20]: 👍

**Pull request merged**